pkg/controllers/dashboard/systemcharts/controller.go

```diff
@@ -571,3 +620,32 @@ func isInHarvesterLocal() bool {
 	}
 	return false
 }
+
+// namespaceGone returns true if the given namespace was deleted
+func (h *handler) namespaceGone(ns string) bool {
+	if ns == "" {
+		return true
+	}
+	_, err := h.namespaceCache.Get(ns)
+	if errors.IsNotFound(err) {
+		return true
+	}
+	if err != nil {
+		logrus.Warnf("[systemcharts] failed to get namespace %s: %v", ns, err)
+	}
+	return false
+}
+
+// onNamespace watches for deletion of namespaces used by Provisioning and Turtles charts.
+func (h *handler) onNamespace(_ string, ns *kcorev1.Namespace) (*kcorev1.Namespace, error) {
```

Comment:
Are we always guaranteed to get delete events for namespaces without a finalizer? I know it's been an issue in the past for other resources, but I'd imagine namespaces are a bit more complicated than regular resources.